module/mempool/stdmap/backend.go

```diff
@@ -213,25 +214,15 @@ func (b *Backend) RegisterEjectionCallbacks(callbacks ...mempool.OnEjection) {
 // reduce will reduce the size of the kept entities until we are within the
 // configured memory pool size limit.
 func (b *Backend) reduce() {
-
 	// we keep reducing the cache size until we are at limit again
-	for len(b.entities) > int(b.limit) {
-
+	// this was a loop, but the loop is now in EjectTrueRandomFast()
+	// the ejections are batched, so this call to eject() may not actually
+	// do anything until the batch threshold is reached (currently 128)
+	if len(b.entities) > int(b.ejectionTrigger) {
 		// get the key from the eject function
-		key, _ := b.eject(b.entities)
-
-		// if the key is not actually part of the map, use stupid fallback eject
-		entity, ok := b.entities[key]
-		if !ok {
-			key, _ = EjectFakeRandom(b.entities)
-		}
-
-		// remove the key
-		delete(b.entities, key)
-
-		// notify callback
-		for _, callback := range b.ejectionCallbacks {
-			callback(entity)
-		}
+		// revert to prior commits if this eject function is not
+		// EjectTrueRandomFast
```

Comment:
I'm not sure I get the meaning of this comment?

Reply:
The logic changed drastically, so if anyone wants to use a different eject function, they'll need to revert to prior logic, and basically re-write all of it.